src/fsharp/ConstraintSolver.fs

```diff
@@ -1996,14 +1996,19 @@ and MustUnifyInsideUndo csenv ndeep trace cxsln ty1 ty2 =
 
 and ArgsMustSubsumeOrConvertInsideUndo (csenv:ConstraintSolverEnv) ndeep trace cxsln isConstraint calledArg (CallerArg(callerArgTy,m,_,_) as callerArg) = 
     let calledArgTy = AdjustCalledArgType csenv.InfoReader isConstraint calledArg callerArg
-    SolveTypSubsumesTypWithReport csenv ndeep  m (WithTrace trace) cxsln calledArgTy callerArgTy 
+    SolveTypSubsumesTypWithReport csenv ndeep m (WithTrace trace) cxsln calledArgTy callerArgTy 
 
 and TypesMustSubsumeOrConvertInsideUndo (csenv:ConstraintSolverEnv) ndeep trace cxsln m calledArgTy callerArgTy = 
     SolveTypSubsumesTypWithReport csenv ndeep m trace cxsln calledArgTy callerArgTy 
 
 and ArgsEquivInsideUndo (csenv:ConstraintSolverEnv) isConstraint calledArg (CallerArg(callerArgTy,m,_,_) as callerArg) = 
     let calledArgTy = AdjustCalledArgType csenv.InfoReader isConstraint calledArg callerArg
-    if typeEquiv csenv.g calledArgTy callerArgTy then CompleteD else ErrorD(Error(FSComp.SR.csArgumentTypesDoNotMatch(),m))
+    if typeEquiv csenv.g calledArgTy callerArgTy then 
+        CompleteD
+    else 
+        ErrorD(Error(FSComp.SR.csArgumentTypesDoNotMatch(
```

Comment:
I believe the fact that this is always called ``InsideUndo`` means that this error is never surfaced to the user e.g. see FilterEachThenUndo